memcached-operator/pkg/controller/memcached/memcached_controller.go

```diff
@@ -109,55 +110,41 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, err
 	}
 
-	// Check if the Deployment already exists, if not create a new one
-	deployment := &appsv1.Deployment{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: memcached.Name, Namespace: memcached.Namespace}, deployment)
-	if err != nil && errors.IsNotFound(err) {
-		// Define a new Deployment
-		dep := r.deploymentForMemcached(memcached)
-		reqLogger.Info("Creating a new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
-		err = r.client.Create(context.TODO(), dep)
-		if err != nil {
-			reqLogger.Error(err, "Failed to create new Deployment.", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
-			return reconcile.Result{}, err
-		}
-		// Deployment created successfully - return and requeue
-		// NOTE: that the requeue is made with the purpose to provide the deployment object for the next step to ensure the deployment size is the same as the spec.
-		// Also, you could GET the deployment object again instead of requeue if you wish. See more over it here: https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler
-		return reconcile.Result{Requeue: true}, nil
-	} else if err != nil {
-		reqLogger.Error(err, "Failed to get Deployment.")
-		return reconcile.Result{}, err
-	}
+	// Update Deployment if it has changed to something other than expected. If the Deployment
+	// doesn't exist it will be created (initial creation is not really a special case).
+	deployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: memcached.Name, Namespace: memcached.Namespace}}
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, deployment, func(ignore runtime.Object) error {
```

Comment:
The `controllerutil.CreateOrUpdate` does much more than one action and the actions of createOrUpdate which are implicit in this name. It is checking/managing the resource as well. So, Is not this solution hurting concepts as cohesion for example?

Reply:
I don't fully understand. Are you saying that `controllerutil.CreateOrUpdate` should not be used, or that it's not a good fit for a scenario like this? For my own understanding, are there other cases where you think it would be more appropriate? It seems like a good use of it to me, and I think it is preferable to the old way -- it's somewhat analogous to the `kubectl apply` concept in my opinion.

Reply:
Hi @grdryn,

Sorry, if I could not make clear enough. However,  I am speaking about your implementation using it and not the framework one. I hope that the comment https://github.com/operator-framework/operator-sdk-samples/pull/77/files#r334772440 can clarify my concerns.